### ISSUE TITLE
Analyser error on a non-UTF8 file

### ISSUE THREAD

**Issue by petermnt** [invertase/dart_custom_lint#69]
**Describe the bug**
The analyser error's on a non-UTF8 file.
I tried to use adapt the globs in `filesToAnalyze` but it crashes before it evaluates that.
`analysis_config.yaml` also doesn't seem to have a way to exclude anything but `.dart` files.

```The request analysis.setContextRoots failed with the following error:
RequestErrorCode.PLUGIN_ERROR
A request throw the exception:FileSystemException(path=/local-path-to/file.png; message=Failed to decode data using encoding 'utf-8')
#0      _PhysicalFile.readAsStringSync (package:analyzer/file_system/physical_file_system.dart:158:7)
#1      _OverlayFile.readAsStringSync (package:analyzer/file_system/overlay_file_system.dart:196:18)
#2      FileSource.contentsFromFile (package:analyzer/src/source/source_resource.dart:64:51)
#3      FileSource.contents (package:analyzer/src/source/source_resource.dart:51:12)
#4      _ClientAnalyzerPlugin._createResolverForFile (package:custom_lint_builder/src/client.dart:781:50)
#5      _ClientAnalyzerPlugin.analyzeFile (package:custom_lint_builder/src/client.dart:656:22)
#6      ServerPlugin.analyzeFiles (package:analyzer_plugin/plugin/plugin.dart:119:13)
<asynchronous suspension>
#7      ServerPlugin.afterNewContextCollection.<anonymous closure> (package:analyzer_plugin/plugin/plugin.dart:86:7)
<asynchronous suspension>
#8      ServerPlugin._forAnalysisContexts (package:analyzer_plugin/plugin/plugin.dart:518:9)
<asynchronous suspension>
#9      ServerPlugin.afterNewContextCollection (package:analyzer_plugin/plugin/plugin.dart:84:5)
<asynchronous suspension>
#10     _ClientAnalyzerPlugin._runOperation (package:custom_lint_builder/src/client.dart:797:14)
<asynchronous suspension>
#11     ServerPlugin.handleAnalysisSetContextRoots (package:analyzer_plugin/plugin/plugin.dart:273:5)
<asynchronous suspension>
#12     _ClientAnalyzerPlugin._runOperation (package:custom_lint_builder/src/client.dart:797:14)
<asynchronous suspension>
#13     ServerPlugin._getResponse (package:analyzer_plugin/plugin/plugin.dart:544:18)
<asynchronous suspension>
#14     ServerPlugin._onRequest (package:analyzer_plugin/plugin/plugin.dart:611:18)
<asynchronous suspension>

at:
#0      _SocketCustomLintServerToClientChannel.sendCustomLintRequest.<anonymous closure> (package:custom_lint/src/v2/server_to_client_channel.dart:385:11)
#1      _$_CustomLintResponseAnalyzerPluginResponse.map (package:custom_lint/src/v2/protocol.freezed.dart:924:34)
#2      _SocketCustomLintServerToClientChannel.sendCustomLintRequest (package:custom_lint/src/v2/server_to_client_channel.dart:379:14)
<asynchronous suspension>
#3      _SocketCustomLintServerToClientChannel.sendAnalyzerPluginRequest (package:custom_lint/src/v2/server_to_client_channel.dart:351:22)
<asynchronous suspension>
#4      Future.wait.<anonymous closure> (dart:async/future.dart:522:21)
<asynchronous suspension>
#5      _SocketCustomLintServerToClientChannel.init (package:custom_lint/src/v2/server_to_client_channel.dart:304:5)
<asynchronous suspension>
#6      CustomLintServer._maybeSpawnCustomLintPlugin (package:custom_lint/src/v2/custom_lint_analyzer_plugin.dart:289:5)
<asynchronous suspension>
#7      CustomLintServer._handleAnalysisSetContextRoots.<anonymous closure> (package:custom_lint/src/v2/custom_lint_analyzer_plugin.dart:264:9)
<asynchronous suspension>
#8      PendingOperation.run (package:custom_lint/src/async_operation.dart:14:14)
<asynchronous suspension>
#9      CustomLintServer._handleRequest (package:custom_lint/src/v2/custom_lint_analyzer_plugin.dart:124:22)
<asynchronous suspension>

The request edit.getAssists failed with the following error:
RequestErrorCode.PLUGIN_ERROR
A request throw the exception:Bad state: Unable to find the context to /Users/peter/.pub-cache/hosted/pub.dev/custom_lint_builder-0.2.0/lib/src/client.dart
#0      AnalysisContextCollectionImpl.contextFor (package:analyzer/src/dart/analysis/analysis_context_collection.dart:143:5)
#1      _ClientAnalyzerPlugin.handleEditGetAssists (package:custom_lint_builder/src/client.dart:354:47)
<asynchronous suspension>
#2      ServerPlugin._getResponse (package:analyzer_plugin/plugin/plugin.dart:564:18)
<asynchronous suspension>
#3      ServerPlugin._onRequest (package:analyzer_plugin/plugin/plugin.dart:611:18)
<asynchronous suspension>

at:
#0      _SocketCustomLintServerToClientChannel.sendCustomLintRequest.<anonymous closure> (package:custom_lint/src/v2/server_to_client_channel.dart:385:11)
#1      _$_CustomLintResponseAnalyzerPluginResponse.map (package:custom_lint/src/v2/protocol.freezed.dart:924:34)
#2      _SocketCustomLintServerToClientChannel.sendCustomLintRequest (package:custom_lint/src/v2/server_to_client_channel.dart:379:14)
<asynchronous suspension>
#3      _SocketCustomLintServerToClientChannel.sendAnalyzerPluginRequest (package:custom_lint/src/v2/server_to_client_channel.dart:351:22)
<asynchronous suspension>
#4      CustomLintServer._handleRequest.<anonymous closure>.<anonymous closure> (package:custom_lint/src/v2/custom_lint_analyzer_plugin.dart:140:17)
<asynchronous suspension>
#5      PendingOperation.run (package:custom_lint/src/async_operation.dart:14:14)
<asynchronous suspension>
#6      CustomLintServer._handleRequest (package:custom_lint/src/v2/custom_lint_analyzer_plugin.dart:124:22)
<asynchronous suspension>
```

**To Reproduce**
* Have a .png file somewhere in a subfolder.
* Run the plugin

**Expected behavior**
File gets ignored

**Comment by rrousselGit** [comment]
Out of curiosity, what are you trying to do?

custom_lint tries to read a string, so that it can obtain line numbers & co for showing lints within the file.
Are you actually trying to show lints within the source of a png?

**Comment by petermnt** [comment]
No not at all. I'd like the file to be ignored.
It's just reading non-dart files in the project too for some reason.

Afaik we have a pretty standard setup. With the analysis_options and pubspec in the root, and a lib and test folder which contains all our dart code.
But the custom plugin looks in all files in all folders, including non-text files.

**Comment by rrousselGit** [comment]
You mean that you have a lint that does _not_ try to analyse pngs, and it's read anyway?

**Comment by kuhnroyal** [comment]
I can confirm this behavior, I have some errors where it incorrectly tries to read PNGs in the test assets.

**Comment by rrousselGit** [comment]
Thanks. I'll check and see what's the cause

**Comment by rrousselGit** [comment]
Should be fixed in 0.2.2
